src/memory.rs

```diff
@@ -39,8 +39,18 @@ pub struct Pointer {
 
 impl Pointer {
     pub fn offset(self, i: isize) -> Self {
+        // FIXME: prevent offsetting ZST ptrs in tracing mode
```

Comment:
Why is this a problem for ZSTs in particular? Isn't offsetting past the end of the allocation bad for all pointers?